converter/src/main/java/gov/cms/qpp/conversion/validate/Checker.java

```diff
@@ -122,6 +125,42 @@ public Checker singleValue(LocalizedError code, String name) {
 		return this;
 	}
 
+	/**
+	 * checks target node for a date that is properly formatted.
+	 *
+	 * @param code that identifies the error
+	 * @param name key of the expected value
+	 * @return The checker, for chaining method calls
+	 */
+	public Checker isValidDate(LocalizedError code, String name) {
+		String date = node.getValue(name);
+		try {
+			String parse = cleanString(date);
+			parse = parse.replace("-", "").replace("/", "");
+			if (parse.length() > DATE_FORMAT.length()) {
+				parse = parse.substring(0, DATE_FORMAT.length());
+			}
+			LocalDate.parse(cleanString(parse), DateTimeFormatter.ofPattern(DATE_FORMAT));
+		}
+		catch (DateTimeParseException e) {
+			details.add(detail(code));
+		}
+		return this;
+	}
+
+	/**
+	 * Remove whitespace and lowercases the string passed in
+	 *
+	 * @param value to be formatted
+	 * @return a cleaned string value
+	 */
+	private String cleanString(String value) {
+		if (value == null) {
+			return "";
+		}
+		return value.trim().toLowerCase();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1449)